pkg/file/file.go

```diff
@@ -57,44 +66,200 @@ func NewFile(format string) (File, error) {
 	return nil, utils.NewErrInvalidSegment(format)
 }
 
-// CreateFile attempts to parse raw metro2 file contents
-func CreateFile(buf []byte) (File, error) {
-	fileFormat, dataType, err := getFileInformation(buf)
+// NewFileFromReader attempts to parse raw metro2 file or json file
+func NewFileFromReader(r io.Reader) (File, error) {
+
+	if r == nil {
+		return nil, errors.New("invalid file reader")
+	}
+
+	dummy := &dummyFile{}
+
+	decoder := json.NewDecoder(r)
+	jsonDecodeErr := decoder.Decode(dummy)
+
+	// reset file reader
+	if _, ok := r.(io.Seeker); ok {
+		r.(io.Seeker).Seek(0, io.SeekStart)
+	}
```

Comment:
Why do we need to reset the reader position?

Reply:
the code is necessary because there isn't method that detect tile type without reading
NewFileFromReader function don't know what input reader is, need to detect json or metro2 file 
So I tried to read first block using json package in first part

metro file reader or json reader need to run after reset file's seek pointer